src/Symfony/Component/Mime/CharacterReader/GenericFixedWidthCharacterReader.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mime\CharacterReader;
+
+/**
+ * Provides fixed-width byte sizes for reading fixed-width character sets.
+ *
+ * @author     Chris Corbyn
+ * @author     Xavier De Cock <xdecock@gmail.com>
+ */
+final class GenericFixedWidthCharacterReader implements CharacterReaderInterface
+{
+    /**
+     * The number of bytes in a single character.
+     *
+     * @var int
+     */
+    private $width;
+
+    /**
+     * Creates a new GenericFixedWidthReader using $width bytes per character.
+     */
+    public function __construct(int $width)
+    {
+        $this->width = $width;
+    }
+
+    public function getCharPositions(string $string, int $startOffset, array &$currentMap, &$ignoredChars): int
+    {
+        $strlen = \strlen($string);
+        // % and / are CPU intensive, so, maybe find a better way
+        $ignored = $strlen % $this->width;
+        $ignoredChars = $ignored ? substr($string, -$ignored) : '';
+        $currentMap = $this->width;
```

Comment:
seems you have a type error here, `currentMap` must be an `array` but it is assigned with an `int`

Reply:
removed the type hint for now, the code is quite messy, so it will need more thinking